toolkit/scripts/toolchain/download_toolchain_rpm.sh

```diff
@@ -178,25 +184,45 @@ function download() {
     return 1
 }
 
+# Validate the signatures of the downloaded RPM against the provided GPG keys
 function validate_signatures() {
     echo "Validating toolchain RPM: $rpm_name" | tee -a "$log_file"
 
-    for key in $allowable_gpg_keys; do
-        echo "Adding key ($key) to empty workdir at ($work_dir)" >> "$log_file"
-        rpmkeys --root "$work_dir" --import "$key" >> "$log_file"
-    done
-
     if ! rpmkeys --root "$work_dir" --checksig --verbose "$dst_file" -D "%_pkgverify_level signature" >> "$log_file"; then
         echo "Failed to validate toolchain package $rpm_name signature, aborting." | tee -a "$log_file"
         return 1
     fi
     return 0
 }
 
-mkdir -p "$(dirname "$log_file")"
+# Validate the integrity of the downloaded RPM using its digests
+function validate_hash() {
+    echo "Checking hash of toolchain RPM: $rpm_name" | tee -a "$log_file"
+
+    if ! rpm --root "$work_dir" --checksig --verbose "$dst_file" -D "%_pkgverify_level digest" >> "$log_file"; then
+        echo "Failed to validate toolchain package $rpm_name hashes, aborting." | tee -a "$log_file"
+        return 1
+    fi
+    return 0
+}
+
+function import_rpm_keys() {
+    echo "Allowing GPG keys" >> "$log_file"
+    for key in $allowable_gpg_keys; do
```

Comment:
similar concern about using the global here.

Reply:
For 3.0 there should only be one key...right?

Reply:
yes, by default we only have 1 key in 3.0. But I think we should still support additional keys in the future if needed.